src/pair.py

```diff
@@ -66,3 +66,76 @@ def __init__(self, nlist, default_r_cut=None, default_r_on=0, mode='none'):
             TypeParameterDict(epsilon=float, len_keys=2),
         )
         self._add_typeparam(params)
+
+
+class PerturbedLennardJones(pair.Pair):
+    r"""Perturbed Lennard-Jones potential.
+
+    Args:
+        nlist (hoomd.md.nlist.NeighborList): Neighbor list.
+        default_r_cut (float): Default cutoff radius :math:`[\mathrm{length}]`.
+        default_r_on (float): Default turn-on radius :math:`[\mathrm{length}]`.
+        mode (str): Energy shifting/smoothing mode.
+
+    :py:class:`PerturbedLennardJones` is a Lennard-Jones perturbation potential.
+    The potential has a purely repulsive (Weeks-Chandler-Andersen) core, with a
+    parameter :math:`\lambda` setting the strength of the attractive tail. When
+    :math:`\lambda` is 0, the potential is purely repulsive. When
+    :math:`\lambda` is 1, the potential is the standard Lennard-Jones potential
+    (see :py:class:`hoomd.md.pair.LJ` for details).
+
+    .. math::
+        :nowrap:
+
+        \begin{eqnarray*}
+        V(r)  = & V_{\mathrm{LJ}}(r, \varepsilon, \sigma) +
+        (1-\lambda)\varepsilon & r < 2^{1/6}\sigma \\
+              = & \lambda V_{\mathrm{LJ}}(
+                r, \varepsilon, \sigma) & 2^{1/6}\sigma \ge
+                  r < r_{\mathrm{cut}} \\
+              = & 0 & r \ge r_{\mathrm{cut}}
+        \end{eqnarray*}
+
+
+    Example::
+
+        nl = nlist.Cell()
+        perturbed_lj = pair.PerturbedLennardJones(default_r_cut=3.0, nlist=nl)
+        perturbed_lj.params[('A', 'A')] = dict(epsilon=1.0, sigma=1.0, lam=0.5)
+        perturbed_lj.r_cut[('A', 'B')] = 3.0
+
+    .. py:attribute:: params
+
+        The Perturbed LJ potential parameters. The dictionary has the following
+        keys:
+
+        * ``epsilon`` (`float`, **required**) - energy parameter
+          :math:`\varepsilon` :math:`[\mathrm{energy}]`
+        * ``sigma`` (`float`, **required**) - particle size :math:`\sigma`
+          :math:`[\mathrm{length}]`
+        * ``lam`` (`float`, **required**) - scale factor for attraction,
+        between 0 and 1 :math:`\lambda`
+          :math:`[\mathrm{dimensionless}]`
```

Comment:
@astatt Should we rename this to something more descriptive? It is `lam` because `lambda` is a reserved keyword in Python. Maybe we should call it `attraction_scale_factor`?
```suggestion
        * ``attraction_scale_factor`` (`float`, **required**) - scale factor for attraction :math:`\lambda`,
          usually between 0 and 1 
          :math:`[\mathrm{dimensionless}]`
```

Reply:
I agree! `attraction_scale_factor` is good!